lib/Table/TableColumn.ts

```diff
@@ -158,47 +158,163 @@ export default class TableColumn {
 
-      let d: Date | null;
-      if (replaceWithNull && replaceWithNull.indexOf(value) >= 0) {
-        d = null;
-      } else if (value.length === 0) {
-        d = null;
-      } else {
-        d = toDate(values[i]);
-        if (d === null) {
-          ++numberOfNonDates;
+    type StringToDateFunction = (str: string) => Date | null;
+
+    const centuryFix = (y: number) =>
+      y < 50 ? 2000 + y : y < 100 ? 1900 + y : y;
+
+    const ddmmyyyy: StringToDateFunction = value => {
+      // Try dd/mm/yyyy and watch out for failures that would also cross out mm/dd/yyyy
+      for (let separator of separators) {
+        const sep1 = value.indexOf(separator);
+        if (sep1 === -1) continue; // Try next separator
+        const sep2 = value.indexOf(separator, sep1 + 1);
+        if (sep2 === -1) {
+          // Neither ddmmyyyy nor mmddyyyy
+          parsingFailed = true;
+          skipMmddyyyy = true;
+          return null;
+        }
+        let dayString = value.slice(0, sep1);
+        let monthString = value.slice(sep1 + 1, sep2);
+        let yearString = value.slice(sep2 + 1);
+        const d = +dayString;
+        const m = +monthString;
+        const y = +yearString;
+        if (Number.isInteger(d) && Number.isInteger(m) && Number.isInteger(y)) {
+          if (d > 31 || y > 9999) {
+            // Neither ddmmyyyy nor mmddyyyy
+            parsingFailed = true;
+            skipMmddyyyy = true;
+            return null;
+          }
+          if (m > 12) {
+            // Probably mmddyyyy
+            parsingFailed = true;
+            return null;
+          }
+          return new Date(centuryFix(y), m - 1, d);
+        } else {
+          // Neither ddmmyyyy nor mmddyyyy
+          parsingFailed = true;
+          skipMmddyyyy = true;
+          return null;
+        }
+      }
+      // Neither ddmmyyyy nor mmddyyyy
+      parsingFailed = true;
+      skipMmddyyyy = true;
+      return null;
+    };
+
+    let mmddyyyy: StringToDateFunction = value => {
+      // This function only exists to allow mm-dd-yyyy dates
+      // mm/dd/yyyy dates could be picked up by `new Date`
+      const separator = "-";
+      const sep1 = value.indexOf(separator);
+      if (sep1 === -1) {
+        parsingFailed = true;
+        return null;
+      }
+      const sep2 = value.indexOf(separator, sep1 + 1);
+      if (sep2 === -1) {
+        parsingFailed = true;
+        return null;
+      }
+      let monthString = value.slice(0, sep1);
+      let dayString = value.slice(sep1 + 1, sep2);
+      let yearString = value.slice(sep2 + 1);
+      const d = +dayString;
+      const m = +monthString;
+      const y = +yearString;
+      if (Number.isInteger(d) && Number.isInteger(m) && Number.isInteger(y)) {
+        if (d > 31 || m > 12 || y > 9999) {
+          parsingFailed = true;
+          return null;
         }
+        return new Date(centuryFix(y), m - 1, d);
+        return new Date(`${centuryFix(y)}/${m}/${d}`);
```

Comment:
There are two returns here

Reply:
and omg months start at 0 - that is very upsetting

Reply:
ikr